packages/neon-core/src/u/misc.ts

```diff
@@ -78,3 +79,39 @@ export function reverseHex(hex: string): string {
   }
   return out;
 }
+
+/**
+ * Calculates the byte size of any supported input following NEO's variable int format.
+ */
+export function getVarSize(value: any): number {
+  if (typeof value === "object" && value instanceof HexString) {

```

Comment:
Structurally, this method will read a lot better if we perform a switch case on `typeof` before narrowing it down. The object case can then be decomposed outside.

Reply:
👍 
coming from python I'm not used to having a `switch` available 😅